_search-plugins/knn/filter-search-knn.md

```diff
@@ -0,0 +1,647 @@
+To run k-NN queries with a filter, the Lucene search engine and HSNW method are required.
+
+To learn more about how to use Query DSL Boolean query clauses, see [Boolean queries]({{site.url}}{{site.baseurl}}/opensearch/query-dsl/bool).
+{: .note }
+
+## How does a k-NN filter work?
+
+The OpenSearch k-NN plugin version 2.2 provided support for the Lucene engine to process k-NN searches. The Lucene engine provides a search that is based on the Hierarchical Navigable Small World (HSNW) algorithm to represent a multi-layered graph. The OpenSearch k-NN plugin version 2.4 is able to incorporate filters for searches based on Lucene 9.4.
+
+After a filter is applied to a set of documents to be searched, the algorithm decides whether to perform pre-filtering for an exact kNN search or modified post-filtering for approximate search. The approximate search with filtering guarantees the top number of closest vectors in result.
+
+Lucene also provides the capability to operate its `KnnVectorQuery` over a subset of documents. To learn more about Lucene’s new capability, see the [Apache Lucene Documentation](https://issues.apache.org/jira/browse/LUCENE-10382).
+
+### Filtered search performance
+
+Filtering that is tightly integrated with the Lucene HNSW algorithm implementation allows you to to apply k-NN searches more efficiently both in terms of relevancy of search results and performance. Consider that if you do an exact search on a large data set, the results are slow, and post-filtering does not guarantee the required number of results you specify for the `k` value.
+With this new capability, you can create an approximate k-NN search and apply filters, with the amount of results you need.
+
+The following workflow diagram shows how the HSNW algorithm decides which type of filtering to apply to a search based on the volume of documents, and number of `k` points in the index that you search with a filter. The variables shown in the diagram are described in the table below.
+
+Variable | Description |
+-- | -- | -- |
+N | Number of documents in the index.
+P | Number of documents in the search set after the filter is applied using the formula: P <= N.
+q | The search vector.
+k | The maximum number of vectors to return in the response.
+
+![How the algorithm evaluates a doc set]({{site.url}}{{site.baseurl}}/images/hsnw-algorithm.png)
+
+***Figure 1: Filter algorithm workflow***
+
+## Filter approaches by use case
+
+Depending on the data set that you are searching, you might choose a different approach to minimize recall or latency. You can create filters that are either: very selective (80%), somewhat selective (38%), or not very selective (2.5%). The selectiveness percentage indicates the amount the filter returns for any given document set in an index.
```

Comment:
now I'm not sure selective is good word to describe the filter. Initially I was thinking that more selective filter means more criteria in filter and as a result less documents returned, but seem that can be interpreted in an opposite way and that makes sense as well. 
Maybe we can change it to something more obvious, like strict or restrictive. More strict filter return less document and vise versa.

Reply:
ok. let me do the rewrites now to change terms throughout as follows: 

very selective -> Restrictive
somewhat selective -> Somewhat restrictive
not very selective -> Not very restrictive

Reply:
I think it should be the other way around:

very selective -> Not very restrictive
somewhat selective -> Somewhat restrictive
not very selective -> very restrictive

so very restrictive filter leaves the smallest number of docs, 2.5 % in our example, and not restrictive filter has very few criteria and thus retrieves the highest number of docs 80% in our example.